src/overlays/actors/ovl_En_Racedog/z_en_racedog.c

```diff
@@ -77,63 +141,532 @@ static DamageTable D_80B25ED0 = {
+            this->actor.speedXZ = 7.5f;
+        }
+    } else {
+        if (this->actor.speedXZ > 7.0f) {
+            this->actor.speedXZ = 7.0f;
+        }
+    }
+}
+
+void func_80B251EC(EnRacedog* this) {
+    f32 temp;
+
+    if (D_80B25D88[this->unk_290].unk_00 == -1.0f) {
+        if (D_80B25D48 < 100.0f) {
+            D_80B25D88[this->unk_290].unk_00 = 200.0f / (200.0f - D_80B25D48);
+        } else {
+            D_80B25D88[this->unk_290].unk_00 = 2.0f;
+        }
+    }
+
+    if (D_80B25D88[this->unk_290].unk_0E < 0x3542) {
+        temp = D_80B25D88[this->unk_290].unk_00;
+        this->unk_298 = temp * D_80B25D50[D_80B25D88[this->unk_290].unk_08][1];
+    }
+
+    if ((D_80B25D88[this->unk_290].unk_0E < 0x353E) && (this->unk_290 != D_80B25D4C)) {
+        this->unk_298 *= D_80B25D88[this->unk_290].unk_04;
+    }
+}
+
+void func_80B252F8(EnRacedog* this) {
+    if ((this->unk_1E8 >= 9) && (this->unk_29C == 0)) {
+        this->unk_29C = 1;
+    }
+
+    if ((this->unk_1E8 >= 0xB) && (this->unk_29C == 1)) {
+        this->unk_29C = 2;
+    }
+
+    if (((this->unk_1E8 >= D_80B25D44) || (this->unk_29C <= 0)) && (this->unk_1E8 > D_80B25D44)) {
+        D_80B25D44 = this->unk_1E8;
+        D_80B25D4C = this->unk_290;
+    }
+}
+
+void func_80B2538C(EnRacedog* this) {
+    if (func_80B25490(this, D_80B25E78) && this->unk_29C == 2) {
+        D_80B25D40++;
+        if (D_80B25D40 == 1) {
+            Audio_QueueSeqCmd(NA_BGM_HORSE_GOAL | 0x8000);
+            play_sound(NA_SE_SY_START_SHOT);
+        }
+
+        this->unk_29C = 3;
+        if (this->unk_290 == this->unk_292) {
+            gSaveContext.eventInf[0] = (gSaveContext.eventInf[0] & 7) | (D_80B25D40 * 8);
+        }
+    }
+}
+
+void func_80B25448(EnRacedog* this) {
+    if (this->actor.speedXZ < 3.0f) {
+        D_80B25EF0[2].playSpeed = 0.9f;
+    } else {
+        D_80B25EF0[2].playSpeed = 1.0f;
+    }
+}
+
+s32 func_80B25490(EnRacedog* this, Vec2f* arg1) {
+    f32 xDistToTopFront;
+    f32 zDistToTopFront;
+    f32 xDistToBottomFront;
+    f32 xDistToBottomBack;
+    f32 zDistToBottomBack;
+    f32 zDistToBottomFront;
+    f32 zDistToTopBack;
+    f32 xDistToTopBack;
+    f32 temp_f0;
+    f32 temp;
+
+    xDistToTopFront = this->actor.world.pos.x - arg1[0].x;
+    zDistToTopFront = this->actor.world.pos.z - arg1[0].y;
+    xDistToBottomFront = this->actor.world.pos.x - arg1[1].x;
+    zDistToBottomFront = this->actor.world.pos.z - arg1[1].y;
+    xDistToBottomBack = this->actor.world.pos.x - arg1[2].x;
+    zDistToBottomBack = this->actor.world.pos.z - arg1[2].y;
+    xDistToTopBack = this->actor.world.pos.x - arg1[3].x;
+    zDistToTopBack = this->actor.world.pos.z - arg1[3].y;
+
+    temp_f0 = ((xDistToTopFront * zDistToBottomFront) - (xDistToBottomFront * zDistToTopFront));
+    temp = (((xDistToBottomFront * zDistToBottomBack) - (xDistToBottomBack * zDistToBottomFront)));
+    if (temp_f0 * temp < 0.0f) {
+        return 0;
```

Comment:
```suggestion
        return false;
```
Seems to return a boolean